notifications/pipeline_steps/base.py

```diff
@@ -24,5 +24,7 @@ def process_message(message):
         logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
         if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
```

Comment:
Nit: We want this method to be implemented by each Pipeline step, correct?  If so, I would want the expiration check to be handled by the `Pipeline.process` method instead - so it is implemented and checked in one place for all pipeline steps.  If you make this change, you'll need to update your test as well.